core/src/main/java/de/danoeh/antennapod/core/feed/util/ImageResourceUtils.java

```diff
@@ -1,50 +1,66 @@
 package de.danoeh.antennapod.core.feed.util;
 
-import de.danoeh.antennapod.core.asynctask.ImageResource;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import de.danoeh.antennapod.core.feed.Feed;
 import de.danoeh.antennapod.core.feed.FeedItem;
-import de.danoeh.antennapod.core.feed.FeedMedia;
 import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.util.playback.Playable;
 
 /**
- * Utility class to use the appropriate image resource based on {@link UserPreferences}
+ * Utility class to use the appropriate image resource based on {@link UserPreferences}.
  */
 public final class ImageResourceUtils {
 
     private ImageResourceUtils() {
     }
 
-    public static String getImageLocation(ImageResource resource) {
+    @Nullable
+    public static String getImageLocation(@NonNull Playable playable) {
 
         if (UserPreferences.getUseEpisodeCoverSetting()) {
-            return resource.getImageLocation();
+            return playable.getImageLocation();
         } else {
-            return getShowImageLocation(resource);
+            return getFallbackImageLocation(playable);
```

Comment:
This conflicts with PR #4908. I think it would be better to merge that one first. In general, I really like that you no longer need `instanceof` here :+1:

Reply:
Thanks, I will wait for #4908 to be merged and update this PR afterwards.

Reply:
#4908 is merged now. Sorry for the merge conflicts. Would be great if you could rebase the PR.

Reply:
Done, I rebased the PR to the head of the develop branch.